scipy/stats/_continuous_distns.py

```diff
@@ -6121,19 +6122,82 @@ def _entropy(self, c):
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return ndata * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the partial derivatives
+                # w.r.t. to location and scale equal.
+                location = np.min(data) - scale
+                shape = fshape or get_shape(scale, location)
+                return dLdLocation(shape, location) - dLdScale(shape, scale)
+
+            def interval_contains_root(lbrack, rbrack):
+                # return true if the signs disagree.
+                return (np.sign(fun_to_solve(lbrack)) !=
+                        np.sign(fun_to_solve(rbrack)))
+
+            # set brackets for `root_scalar` to use when optimizing over the
+            # scale such that a root is likely between them.
+            lbrack, rbrack = .1, np.max(data)
+            # if a root is not between the brackets, iteratively expand them
+            # until they include a sign change, checking after each bracket is
+            # modified.
+            while not interval_contains_root(lbrack, rbrack):
+                lbrack /= 2
+                rbrack *= 2
+            res = root_scalar(fun_to_solve, bracket=[lbrack, rbrack])
+
+            if res.converged:
+                fscale = res.root
+                floc = np.min(data) - fscale
+                fshape = fshape or get_shape(fscale, floc)
+
+                # The Pareto distribution requires that its parameters satisfy
+                # the condition `fscale + floc <= min(data)`. However, to
+                # avoid numerical issues, we require that `fscale + floc`
+                # is strictly less than `min(data)`. If this condition
+                # is not satisfied, reduce the scale with `np.nextafter` to
+                # ensure that data does not fall outside of the support.
+                if not (fscale + floc) < np.min(data):
+                    fscale = np.min(data) - floc
+                    fscale = np.nextafter(fscale, 0)
+                return fshape, floc, fscale
+            else:
+                return super().fit(data, **kwds)
+        elif floc is None:
+            floc = np.min(data) - fscale
+
         if np.any(data - floc < (fscale if fscale else 0)):
             raise FitDataError("pareto", lower=1, upper=np.inf)
```

Comment:
> So this can only happen in the two remaining cases - the ones we haven't already dealt with.
> 
> If the user passes floc and fscale
> If the user passes floc but not fscale

moving to top works well